crates/rolldown_plugin/src/worker_manager.rs

```diff
@@ -0,0 +1,69 @@
+use async_channel::{Receiver, Sender};
+
+pub(crate) struct WorkerManager {
+  free_workers_sender: Sender<u16>,
```

Comment:
This is probably enough as a prototype. However, since you could pass the value around, you could actually think of a way to pass `PluginOptions` or mutiple `PluginOptions` directly to the bundler. That's would be more friednly to interact with `ThreadSafePlugin`.

Reply:
Would you expand this a bit more?

Reply:
Yeah. This is actually a rough thought. Let me see if I could make it clear.

1. For ever  js `rolldown(...)` call, there is a uuid called `BundlerId` generated in js size or other values that's unique.
2. In js, in the calling of `write` or `generate`, when you normalize the plugins in js side, initialize these worker plugin with passing the `BundlerId` and a unique generated `WorkerId` for each worker.
3. Create the rust bundler with options and `BundlerId`
4. While the rust bundler is creating, follwing rust code would be executed
```rust
// Not runnable but you get the idea.
static BundlerToWorkerMap: DashMap<BundlerId, DashMap<WokerId, Vec<PluginOptions>>>

struct WokerPlugin(BundlerId, WorkerId, Option<Vec<PluginOptions>>);

impl Plugin for WokerPlugin {
  async fn build_start() {
     let plugin_options = BundlerToWorkerMap.get().get().expect("Failed to get worker plugins")
     if pluin_options.len == 1 {
       // single mode
       self.2 = Some(plugin_options)
     } else {
       plugin_options = map(option -> Mutex::new(option))
     }
  }

 async fn transform() {
    for p in self.2 {
      p.transform()
     }
  }
}

fn new_bundler(plugins: Vec<PluginOptionsOrBundlerIdWithWorkerId>) {
  // ...
let plugins: Vec<dyn Plugin>  plugins.map(|item| if item is PluginOptions { JsPlugn::new(item) else { WorkerPlugin(item.0, item1) } }).
}

#[napi]
pub fn registerWorkerPlugin(bundlerId, workerId, plugin_options: Vec<PluginOptions>) {
  BundlerToWorkerMap.get_or_insert_default(bundlerId).get_or_insert_default().insert(WorkerId)
}
```

5. in js's worker code, do
```js
import { registerWokerPlugin } from 'rolldown'

// single mode
registerWokerPlugin(bundlerId, workerId, {
 name: 'signle',
 transform: () => {
  }
})
// mutiple
registerWokerPlugin(bundlerId, workerId, 
[
{
 name: 'signle',
 transform: () => {
  }
},
{
 name: 'signle2',
 transform: () => {
  }
},
])

```

Reply:
How we free these resource if something worng would be a issue. However, I haven't tackle them, but I guess it won't be a big problem.

Reply:
I think I got what you mean. What would be the upside of doing this?

Reply:
I'm seeing the new changes and you probably already did what I mean.